bpython/line.py

```diff
@@ -34,7 +34,33 @@ def current_word(cursor_offset: int, line: str) -> Optional[LinePart]:
     return LinePart(start, end, word)
 
 
-_current_dict_key_re = LazyReCompile(r"""[\w_][\w0-9._]*\[([\w0-9._(), '"]*)""")
+# pieces of regex to match repr() of several hashable built-in types
+_match_identifiers_and_functions = r"""[a-zA-Z_][a-zA-Z0-9_.(),\[\] ]*"""
+_match_single_quote_str_bytes = \
+    r"""b?'(?:\\(?:\\\\)*['"nabfrtvxuU]|[^'\\])*(?:\\\\)*\\?'?"""
+_match_double_quote_str_bytes = r"""b?"[^"]*"?"""
+_match_int_and_float = r"""\-?\d+(?:\.\d*)?|\-"""
+_match_str_bytes_int_float = \
+    f'(?:{_match_single_quote_str_bytes}|{_match_double_quote_str_bytes}|' \
+    f'{_match_int_and_float})'
+_match_tuple = r"""\((?:""" \
+    f'{_match_str_bytes_int_float}' \
+    r""", )*(?:""" \
+    f'{_match_str_bytes_int_float}' \
+    r"""\,|""" \
+    f'{_match_str_bytes_int_float}' \
+    r""")?\)?"""
+
+_current_dict_key_re_base = r"""[\w_][\w0-9._]*\["""
+_current_dict_key_re = LazyReCompile(
+    f'{_current_dict_key_re_base}('
```

Comment:
the "_base" part of _current_dict_key_re_base confuses me a little, _base because it's the beginning of the pattern? because the dict is the base of the expression?

Reply:
Yea, I could use some direction on the name if you don't like that, but basically, `base` is the part that I did not change which is everything before the `[` character after which 'key matching mode' is entered.

Reply:
maybe `_current_dict_key_dict_re`, or maybe since all of these substrings are not indented to be used separately, a different name like `_current_dict_key_dict_part` — OH or to match your current convention of `_match_*` maybe just `_dict` or `_dict_before_match` or something. 🤞 there'll be a beautiful was to do this with re.VERBOSE, but if not just changing to something shorter is fine. I just don't want to force readers to know the history of the code in order to understand the variable name.

Reply:
Renamed to `_match_dict_before_key`.